services/fluxmonitor_test.go

```diff
@@ -56,73 +62,162 @@ func (sm *mockBlockchainManager) CreateJobRun(t string, roundId uint32) (map[str
 
 	return nil, errors.New("job run type not implemented")
 }
+
+func createMockAdapter(result string) *url.URL {
+	var payload []byte
+	var statusCode int
+	if result != "" {
+		statusCode = 200
+		payload = []byte(fmt.Sprintf(`{"jobRunID": "1", "result": "%s", "statusCode": %d}`, result, statusCode))
+	} else {
+		statusCode = 400
+		payload = []byte(fmt.Sprintf(`{"jobRunID": "1", "status": "errored", "statusCode": %d}`, statusCode))
+	}
+
+	mockAdapter := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(statusCode)
+		_, _ = w.Write([]byte(fmt.Sprint(bytes.NewBuffer(payload))))
```

Comment:
Oh just noticed this. Any way this doesn't work?

```suggestion
		_, _ = w.Write(payload)
```